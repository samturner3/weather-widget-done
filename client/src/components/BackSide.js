import React from 'react';
import PropTypes from 'prop-types';
import Autocomplete from 'react-google-autocomplete';
import CacheOptions from './CacheOptions';

const BackSide = ({
	setLocation,
	placesCache
}) => {
	return (
		<div>
			<h2>Search a location to show data</h2>
			<Autocomplete
				style={{width: '90%'}}
				onPlaceSelected={(place) => {
					if (place.geometry !== undefined) {
						setLocation(place, place.geometry.location.lat(), place.geometry.location.lng());
					}
				}}
				types={['(regions)']}
			/>
			{(placesCache !== null && placesCache.length > 0) &&
			<div>
				<h3>Server-side cached data locations:</h3>
				<CacheOptions placesCache={placesCache} />
			</div>
			}
		</div>
	);
};

BackSide.propTypes = {
	setLocation: PropTypes.func.isRequired,
	placesCache: PropTypes.arrayOf(PropTypes.shape({
		exp: PropTypes.number.isRequired,
		key: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired
	}))
};

export default BackSide;
