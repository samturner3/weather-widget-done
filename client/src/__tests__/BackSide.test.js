import React from 'react';
import BackSide from '../components/BackSide';

import { selectedPlaceMock } from './__mocks__/mockData'

function setup() {
	const props = {
		setLocation: jest.fn(),
		placesCache: [
			{
				exp: 1535100127158,
				key: '-33.8688197+151.20929550000005',
				name: 'Sydney'
			}
		]
	};

	return {
		wrapper: shallow(<BackSide {...props} />),
		props
	};
}

describe('<BackSide />', () => {
	it('should render all elements', () => {
		const { wrapper } = setup();
		expect(wrapper).toMatchSnapshot();

		// imported component test
		expect(wrapper.find('CacheOptions')).toHaveLength(1);
		expect(wrapper.find('ReactGoogleAutocomplete')).toHaveLength(1);
	});

	describe('Autocomplete\'s onPlaceSelected()', () => {
		it('should call setLocation() with 3 arguments when clicked', () => {
			const { wrapper, props } = setup();
			const setLocationSpy = jest.spyOn(props, 'setLocation');

			wrapper.find('ReactGoogleAutocomplete').props().onPlaceSelected(selectedPlaceMock);
			expect(setLocationSpy)
				.toHaveBeenCalledWith(
					selectedPlaceMock,
					selectedPlaceMock.geometry.location.lat(),
					selectedPlaceMock.geometry.location.lng()
				)
		});
	})
});
