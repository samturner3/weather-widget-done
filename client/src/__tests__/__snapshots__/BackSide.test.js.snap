// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackSide /> should render all elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackSide
    placesCache={
        Array [
            Object {
              "exp": 1535100127158,
              "key": "-33.8688197+151.20929550000005",
              "name": "Sydney",
            },
          ]
    }
    setLocation={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Search a location to show data
</h2>,
        <ReactGoogleAutocomplete
          onPlaceSelected={[Function]}
          style={
                    Object {
                              "width": "90%",
                            }
          }
          types={
                    Array [
                              "(regions)",
                            ]
          }
/>,
        <div>
          <h3>
                    Server-side cached data locations:
          </h3>
          <CacheOptions
                    placesCache={
                              Array [
                                        Object {
                                          "exp": 1535100127158,
                                          "key": "-33.8688197+151.20929550000005",
                                          "name": "Sydney",
                                        },
                                      ]
                    }
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search a location to show data",
        },
        "ref": null,
        "rendered": "Search a location to show data",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onPlaceSelected": [Function],
          "style": Object {
            "width": "90%",
          },
          "types": Array [
            "(regions)",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Server-side cached data locations:
</h3>,
            <CacheOptions
              placesCache={
                            Array [
                                          Object {
                                            "exp": 1535100127158,
                                            "key": "-33.8688197+151.20929550000005",
                                            "name": "Sydney",
                                          },
                                        ]
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Server-side cached data locations:",
            },
            "ref": null,
            "rendered": "Server-side cached data locations:",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "placesCache": Array [
                Object {
                  "exp": 1535100127158,
                  "key": "-33.8688197+151.20929550000005",
                  "name": "Sydney",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Search a location to show data
</h2>,
          <ReactGoogleAutocomplete
            onPlaceSelected={[Function]}
            style={
                        Object {
                                    "width": "90%",
                                  }
            }
            types={
                        Array [
                                    "(regions)",
                                  ]
            }
/>,
          <div>
            <h3>
                        Server-side cached data locations:
            </h3>
            <CacheOptions
                        placesCache={
                                    Array [
                                                Object {
                                                  "exp": 1535100127158,
                                                  "key": "-33.8688197+151.20929550000005",
                                                  "name": "Sydney",
                                                },
                                              ]
                        }
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search a location to show data",
          },
          "ref": null,
          "rendered": "Search a location to show data",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPlaceSelected": [Function],
            "style": Object {
              "width": "90%",
            },
            "types": Array [
              "(regions)",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Server-side cached data locations:
</h3>,
              <CacheOptions
                placesCache={
                                Array [
                                                Object {
                                                  "exp": 1535100127158,
                                                  "key": "-33.8688197+151.20929550000005",
                                                  "name": "Sydney",
                                                },
                                              ]
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Server-side cached data locations:",
              },
              "ref": null,
              "rendered": "Server-side cached data locations:",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "placesCache": Array [
                  Object {
                    "exp": 1535100127158,
                    "key": "-33.8688197+151.20929550000005",
                    "name": "Sydney",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
